### PR TITLE
Update API v58.0 to v59.0

### LINKED ISSUE #1396: RollupService fails on Nonprofit Cloud's standard 'Program' object
**Describe the bug**
When rolling up from a custom child object to the Nonprofit Cloud's standard 'Program' object, either via the child trigger or using the scheduled rollup job, the following Apex error occurs:

  dlrs.RollupService.triggerHandler(Schema.SObjectType)
  FATAL_ERROR|System.QueryException: sObject type 'Program' is not supported.

**To Reproduce**

1. Create an org with the new Nonprofit Cloud and install DLRS. There will be a new standard object called 'Program'.
2. Create a custom child object with a lookup to the standard 'Program' object. 
3. In DLRS, create a simple rollup that counts child records and saves to a number field on the parent. Make it a 'Realtime' rollup, install the child trigger, and activate.
4. Add a new Program record, and then add a child record.
5. See the following error message:

  System.QueryException: sObject type 'Program' is not supported.

**Expected behavior**
This rollup works when running "Full Calculate" - it should also work with Realtime triggered calculations or scheduled rollups.

Screenshot 2023-10-16 at 1 21 21 PM

**Additional context**
The rollup works perfectly when we click Full Calculate - in other words, there's no problem with the aggregate query that is used by the rollup.  The error occurs in a subsequent step - I assume it is when the DLRS code in RollupService tried to query or update the Program object.

Here is the actual query that it runs to do the rollup:
SELECT Target_Program__c, Count(Id) lre0 FROM Referral__c WHERE Target_Program__c in ('11WDp000000L04lMAC') AND (Referral_Date__c = LAST_N_DAYS:365) GROUP BY Target_Program__c

DLRS is running in v58.0 - I tried running my trigger in both 58.0 and 59.0 with the same result.  I also tried running with User and System sharing mode with the same result.

Note that I also tried changing Calculation Mode to Scheduled. When the rollup service ran on schedule, it hit the same error as the triggered rollup.  However, the Full Calculate still works fine.

I have no way of figuring out where the error occurs in the package, but in general I suspect two possible issues with the Nonprofit Cloud 'Program' object. 
1) Since this is a new object, it is possible that some Apex functionality isn't working properly in v58.0 (or maybe not working properly at all). 
2) My user has all the appropriate licenses and permsets to access the Program object, but it is possible that the Apex code is unable to use those objects or fields for some reason related to security or licensing.

### PR DESCRIPTION
**Pull request by groundwired**
# Critical Changes

updated API version from v58.0 to v59.0 throughout

# Issues Closed

fixes #1396 - dmlGuard files were still at v56.0 which causes problems with new standard objects

### CODE REVIEW

**Comment by groundwired** [comment]
@sfenton3 Moved PR from fork into repo

**Review by aheber — APPROVED** [review]

**Comment by aheber** [comment]
@SFDO-Community/sfdo-release-engineers can we get an approval on the API version bump?

**Comment by aheber** [comment]
Thanks, @groundwired! Appreciate you creating a new PR for this. Once we get final approvals then this can merge and will included in the next release.

**Review by davidmreed — APPROVED** [review]

**Pull request merged**
